sql/2022/interop/backdrop.sql

```diff
@@ -0,0 +1,48 @@
+#standardSQL
+CREATE TEMPORARY FUNCTION countSelectors(css STRING)
+RETURNS INT64 LANGUAGE js
+OPTIONS (library = "gs://httparchive/lib/css-utils.js")
+AS '''
+try {
+  function compute(ast) {
+    let ret = 0;
+
+    walkSelectors(ast, selector => {
+      if (/::backdrop/.test(selector)) {
+        ret++;
+      }
+    });
+
+    return ret;
+  }
+
+  const ast = JSON.parse(css);
+  return compute(ast);
+} catch (e) {
+  return 0;
+}
+''';
+
+WITH detections AS (
+  SELECT
+    client,
+    page,
+    SUM(countSelectors(css)) > 0 AS detected
+  FROM
+    `httparchive.almanac.parsed_css`
```

Comment:
Does the `parsed_css` table contain one row per page? i.e. there is no need to `JOIN` with `httparchive.summary_pages.2022_07_01_*` to get the totals?

Reply:
No but this CTE is grouped by page, so there will always be one row per page in the subsequent query